### PR TITLE
Fix "Data Fetch Error" on product page in offline mode (#2489)

### LINKED ISSUE #2489: [bug]: Disable Apollo RetryLink when offline ?
**Describe the bug**
When we are offlline and when we're trying to open some cached page then we will get the cached page, but after this ApolloRetryLink is trying to call some requests and after then we get Data Fetch Error

**To reproduce**
1. Being online go to some product page 
2. Switch browser to offline mode 
3. Open the same product page (must be cached)
4. You see product page
5. In the background, you are able to see graphql retry requests 
6. After some tries you will see "Data Fetch Error" instead of product page 

**Expected behavior**
Should we use apollo-retry link when client is being offline ?

**Screenshots**

ezgif-4-2d2e07fceced
ezgif-4-5a490eebc7bf

**Additional context**
The newest version of develop branch, but also visible on master 

**Possible solutions**
Use apollo retry link only when client is being offline ? 

**Please complete the following device information:**
 - Device  Mac
 - OS The newest one
 - Browser Chrome, safari
 - Browser Version the newest one
 - Magento Version: doesn't matter 
 - PWA Studio Version: develop / 6.01 
 - NPM version `npm -v`:  doesn't matter 
 - Node Version `node -v`: doesn't matter 

**Please let us know what packages this bug is in regards to:**
- [x] `venia-concept`
- [ ] `venia-ui`
- [ ] `pwa-buildpack`
- [x] `peregrine`
- [ ] `pwa-devdocs`
- [ ] `upward-js`
- [ ] `upward-spec`
- [ ] `create-pwa`

### PR DESCRIPTION
**Pull request by mercs600**
## Description
Initially I was thinking the wrong behavior ( check videoscreen below)  is caused by Apollo Retry Link and generally it is, but we have to change condition to handle it. 

### Scenario: 
1. Being online go to some product page
2. Switch browser to offline mode
3. Open the same product page (must be cached)
4. You see product page
5. In the background, you are able to see graphql retry requests (retry link)
6. After some tries you will see "Data Fetch Error" instead of product page

Check the current code: 
```js
    if (loading && !product) return fullPageLoadingIndicator;
    if (error) return <div>Data Fetch Error</div>;

    if (!product) {}
```

It works when you are online, but when you are offline this is not good, because when you got the `error` though you have product data then you won't display it, because of that error which is caused by retry link.

Being offline you have product data from apollo cache and you want to display it though you get errors from retrylink

So the change is small but needed for us because of PWA purposes. 

## Related Issue
Closes #2489 

## Acceptance 
Just go trough my scenario. 
### Verification Stakeholders
@sirugh 

### Verification Steps
1. Being online go to some product page
2. Switch browser to offline mode
3. Open the same product page (must be cached)
4. You see product page
5. In the background, you are able to see graphql retry requests (retry link)
6. After some tries you should see product page still.

### Wrong behavior: 
ezgif-4-2d2e07fceced

### Good behavior: 
ezgif-4-a69f7dd662f9

### CODE REVIEW

**Comment by devops-pwa-codebuild** [comment]
**Performance Test Results**

The following fails have been reported by WebpageTest. These numbers indicates a possible performance issue with the PR which requires further manual testing to validate.

https://pr-2490.pwa-venia.com : LH Performance Expected 0.85 Actual 0.58, LH Best Practices Expected 1 Actual 0.92
https://pr-2490.pwa-venia.com/venia-tops.html : LH Performance Expected 0.75 Actual 0.36, LH Best Practices Expected 1 Actual 0.92
https://pr-2490.pwa-venia.com/valeria-two-layer-tank.html : LH Performance Expected 0.8 Actual 0.5, LH Accessibility Expected 0.9 Actual 0.89, LH Best Practices Expected 1 Actual 0.92

**Comment by mercs600** [comment]
> **Performance Test Results**
> 
> The following fails have been reported by WebpageTest. These numbers indicates a possible performance issue with the PR which requires further manual testing to validate.
> 
> https://pr-2490.pwa-venia.com : LH Performance Expected 0.85 Actual 0.58, LH Best Practices Expected 1 Actual 0.92
> https://pr-2490.pwa-venia.com/venia-tops.html : LH Performance Expected 0.75 Actual 0.36, LH Best Practices Expected 1 Actual 0.92
> https://pr-2490.pwa-venia.com/valeria-two-layer-tank.html : LH Performance Expected 0.8 Actual 0.5, LH Accessibility Expected 0.9 Actual 0.89, LH Best Practices Expected 1 Actual 0.92

I don't think so that my change decreased these numbers 😄

File: packages/venia-ui/lib/RootComponents/Product/product.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ const Product = () => {
     const { error, loading, product } = talonProps;
 
     if (loading && !product) return fullPageLoadingIndicator;
-    if (error) return <div>Data Fetch Error</div>;
+    if (error && !product) return <div>Data Fetch Error</div>;
```
Comment (sirugh) [inline packages/venia-ui/lib/RootComponents/Product/product.js:33]: Nice find, thanks!

Comment (mercs600) [reply packages/venia-ui/lib/RootComponents/Product/product.js:33]: I added 3 scenarios for tests
1. when only `product` data exist,
2. when `product` and `error` data exist
3. when only `error` data  exist

**Comment by mercs600** [comment]
@sirugh how can we push this PR forward ?

**Comment by sirugh** [comment]
> how can we push this PR forward ?

Normally, contributions go through a prioritization process where our product manager and the team discuss the value, complexity, and effort needed to take a contribution across the finish line. In this case, I've raised this PR as one of high value with minimal effort, which means the wheels are already turning!

Unfortunately our process is not transparent to those outside of our organization but rest assured this PR has eyes on it. We've got a release scheduled so we're all a bit focused on finishing up work for that but once we're done I would expect this to get more attention.

**Comment by dpatil-magento** [comment]
QA Approved.

@mercs600 Can you pls sign `Adobe CLA Signed` check?

**Comment by mercs600** [comment]
@sirugh got it, thank you for informing me. @dpatil-magento I have already signed Adobe CLA.

**Pull request merged**